Contributing.md

```diff
@@ -0,0 +1,137 @@
+
+This repo follows the [Contributor Covenant](http://contributor-covenant.org/version/1/3/0/) Code of Conduct.
+
+This repo uses automated tools to standardize the formatting of code, text files and
+commits.
+ - [Pre-commit hooks](#pre-commit-hooks) validate and automatically apply code
+   formatting rules.
+ - [commitlint](#commitlint) is used as a commit message hook to validate that
+   commit messages follow the convention.
+
+## Pre-commit hooks
+
+The ReadAlong Studio team has agreed to systematically use a number of pre-commit hooks to
+normalize formatting of code. You need to install and enable pre-commit to have these used
+when you do your own commits.
+
+Pre-commit hooks enabled:
+- check-yaml validates YAML files
+- end-of-file-fixer makes sure each text file ends with exactly one newline character
+- trailing-whitespace removes superfluous whitespace at the end of lines in text files
+- [Flake8](https://flake8.pycqa.org/) enforces good Python style rules; more info about
+  using Flake8 in pre-commit hooks at:
+  [Lj Miranda flake8 blog post](https://ljvmiranda921.github.io/notebook/2018/06/21/precommits-using-black-and-flake8/)
+- [isort](https://pycqa.github.io/isort/) orders python imports in a standard way
+- [Black](https://github.com/psf/black), the Uncompromising Code Formatter, refortmats all
+  Python code according to very strict rules we've agreed to follow; more info about Black
+  formatting rules in
+  [The Black code style](https://black.readthedocs.io/en/stable/the_black_code_style.html)
+- [mypy](http://mypy-lang.org/) runs type checking for any statically-typed Python code in
+  the repo
+
+### Enabling pre-commit hooks
+
+All the pre-commit hooks are executed using a tool called
+[pre-commit](https://pre-commit.com/). Once you enable pre-commit, it will run all the
+hooks each time you try to commit anything in this repo.
+
+We've listed all the developper dependencies for the project in
+[requirements.dev.txt](requirements.dev.txt) to make them easy to install:
+
+```sh
+pip install -r requirements.dev.txt
+pre-commit install
+```
+
+Note that you have to run the second command in every Studio sandbox you create, so please
+don't forget to do so when you clone a new sandbox!
+
+## commitlint
+
+The team has also agreed to use commitlint-style commit messages. Install and enable
+[commitlint](https://github.com/conventional-changelog/commitlint) to have your commits
+validated systematically.
+
+Commitlint commits look like this:
+
+    type(optional-scope): subject (i.e., short description)
+
+    optional body, which is free form
+
+    optional footer
+
+Valid types: (these are the default, which we're using as is for now)
+ - build: commits for the build system
+ - chore: maintain the repo, not the code itself
+ - ci: commits for the continuous integration system
+ - docs: adding and changing documentation
+ - feat: adding a new feature
+ - fix: fixing something
+ - improvement: improve something
```

Comment:
I don't have a specific use case, so I don't mind taking it out, but I think we have it right now, because `commit -m'foo: bar'` gives me this error message:
```
✖   type must be one of [build, chore, ci, docs, feat, fix, improvement, perf, refactor, revert, style, test] [type-enum]
```
with `improvement` included in the list.

Reply:
Hum, I see that I have a `commitlint.config.js` file in my sandbox, but I don't remember where it came from. It doesn't include "improvement", but for some reason this config file is ignored and doesn't get taken into account.

Reply:
yeah, that's strange, where is it getting that from?? in package.json it clearly shows it's extending https://www.npmjs.com/package/@commitlint/config-conventional